src/ert/_c_wrappers/enkf/ert_config.py

```diff
@@ -269,7 +279,11 @@ def read_user_config(
     ):
         site_config = cls.read_site_config(use_new_parser=use_new_parser)
         if use_new_parser:
-            return lark_parse(user_config_file, site_config)
+            return lark_parse(
+                file=user_config_file,
+                schema=init_user_config_schema(),
+                site_config=site_config,
+            )
```

Comment:
I think the `except ValueError` clause at 303 will catch any errors in casting it to an int, so that on 310 it is guaranteed to be an int, which is hopefully enough to assume things will be OK on L310.